src/derivation.test.ts

```diff
@@ -166,24 +174,48 @@ describe('derivation', () => {
   describe('bip32Derive', () => {
     it('derives the expected keys and addresses', async () => {
       // generate parent key
-      let parentKey: Buffer;
+      let privateKey: Buffer;
+      let chainCode: Buffer;
 
       /* eslint-disable require-atomic-updates */
-      parentKey = await bip39Derive(mnemonic);
-      parentKey = await bip32Derive(`44'`, parentKey);
-      parentKey = await bip32Derive(`60'`, parentKey);
-      parentKey = await bip32Derive(`0'`, parentKey);
-      parentKey = await bip32Derive(`0`, parentKey);
+      [privateKey, , chainCode] = await bip39Derive(mnemonic);
+      [privateKey, , chainCode] = await bip32Derive(
+        `44'`,
+        privateKey,
+        undefined,
+        chainCode,
+      );
+
+      [privateKey, , chainCode] = await bip32Derive(
+        `60'`,
+        privateKey,
+        undefined,
```

Comment:
I wonder if we can simplify the overall API so we don't have to have so many `undefined` params, thoughts?

Reply:
We could change it to use options objects, rather than passing everything as a parameter? Or do you have another suggestion for this?

Reply:
Yeah, that's what I was thinking. But up to you whether you think it improves things or complicates them too much

Reply:
We can definitely do this, but might be better to do it as part of another PR? Since this is already a pretty big refactor, and I want to avoid refactoring even more things.